pkg/kvstore/allocator/allocator.go

```diff
@@ -518,6 +533,8 @@ func (k *kvstoreBackend) RunGC(ctx context.Context, rateLimit *rate.Limiter, sta
 		}
 	}
 
+	opmetrics.IdentityGCSize.WithLabelValues("alive").Set(float64(gcResult.alive))
```

Comment:
I would essentially like these metrics initialization to be done outside of the `allocator`. One simple approach I see is to return the `gcStats` from `RunGC` and then the metrics can be populated in the caller. For this case in `operator/identity_gc.go`. This will also remove the Prometheus server listen dependency from the tests.
Also, we can then add another label that can export the number of times GC ran and how many of those it failed.

Thoughts @aanm @ArthurChiao ?

Reply:
That's a good approach.

Reply:
> I would essentially like these metrics initialization to be done outside of the allocator. One simple approach I see is to return the gcStats from RunGC and then the metrics can be populated in the caller. For this case in operator/identity_gc.go.

Nice point, I'll have a try.

> Also, we can then add another label that can export the number of times GC ran and how many of those it failed.

@fristonio  Should this be exported as another metric? e.g. 

```
# Reusing the same labels & values in pkg/metrics/metrics.go

cilium_operator_identity_gc_runs_total{outcome='sucesss'} 200
cilium_operator_identity_gc_runs_total{outcome='fail'} 0
```